asyncbox/plugins/sqlalchemy/plugin.py

```diff
@@ -20,30 +20,30 @@ class SQLAlchemyPlugin(BasePlugin):
     """SQLAlchemy ORM plugin."""
 
     settings_class = Settings
-    _session: Optional[AsyncSession]
     engine: Optional[AsyncEngine]
 
     async def on_startup(self, *args: Any, **kwargs: Any) -> None:
         """Startup hook."""
         self.engine = create_async_engine(
             make_url_async(self.settings.POSTGRES_DSN), echo=self.settings.SQL_DEBUG
         )
-
-        make_session = sessionmaker(
+        self._make_session = sessionmaker(
             self.engine, expire_on_commit=False, class_=AsyncSession
         )
-        self._session = make_session()
 
-    async def on_shutdown(self, *args: Any, **kwargs: Any) -> None:
-        """Shutdown hook."""
-        if self.session is not None:
-            await self.session.close()
+    @property
+    def make_session(self) -> Callable:
+        """Return an SQLAlchemy session instance."""
+        if getattr(self, "_make_session", None) is None:
+            raise RuntimeError("Plugin not yet initialized!")
+        return self._make_session
 
     async def healthcheck(self) -> HealtCheckData:
         """Healthcheck function."""
-        try:
-            async with self.session.begin():
-                rows = await self.session.execute("select version()")
+        try:  # noqa: WPS229
+            session = self.make_session()
```

Comment:
You can use:
```async with self.make_session() as session, session.begin():```

[link to docs](https://docs.sqlalchemy.org/en/14/orm/session_basics.html?highlight=session#framing-out-a-begin-commit-rollback-block)

Reply:
done